### PR TITLE
Extract metadata from assembly

### PR DESCRIPTION
**Pull request by qinezh**

### CODE REVIEW

**Comment by vwxyzh** [comment]
#842

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/CompilationUtility.cs [hunk 1]
```diff
@@ -54,6 +56,35 @@ public static Compilation CreateCompilationFromVBCode(string code)
...
+                //TODO: "mscorlib" should be ignored while extracting metadata from .NET Core/.NET Framework
+                paths.Add(typeof(object).Assembly.Location);
+                var assemblies = (from path in paths
+                                  select MetadataReference.CreateFromFile(path)).ToList();
+                var complilation = CS.CSharpCompilation.Create("EmptyProjectWithAssembly", new SyntaxTree[] { }, assemblies);
+                return complilation;
+            }
+            catch (Exception e)
+            {
+                Logger.Log(LogLevel.Warning, $"Error generating compilation from assemblies {string.Join(Environment.NewLine, assemblyPaths)}: {e.Message}. Ignored.");
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/CompilationUtility.cs]: log error?

Comment (qinezh) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/CompilationUtility.cs]: I'd like to keep the same with `CreateCompilationFromCsharpCode`

---
In reply to: [95289758](https://github.com/dotnet/docfx/pull/1131#discussion_r95289758) [](ancestors = 95289758)

Comment (hellosnow) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/CompilationUtility.cs]: but CreateCompilationFromCsharpCode return null while here will throw?

---
In reply to: [95290045](https://github.com/dotnet/docfx/pull/1131#discussion_r95290045) [](ancestors = 95290045,95289758)

Comment (qinezh) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/CompilationUtility.cs]: yes, you're right, I should return null here.

---
In reply to: [95291502](https://github.com/dotnet/docfx/pull/1131#discussion_r95291502) [](ancestors = 95291502,95290045,95289758)

**Comment by 928PJY** [comment]
Question: shuld the .dll files in test folder be ignored?

**Comment by vwxyzh** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

**Comment by qinezh** [comment]
no, UT needs it.

---
In reply to: [271480192](https://github.com/dotnet/docfx/pull/1131#issuecomment-271480192) [](ancestors = 271480192)

**Comment by 928PJY** [comment]
:shipit:

**Pull request merged**